app/components/widgets.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Projects4Me Copyright (c) 2017. Licensing : http://legal.projects4.me/LICENSE.txt. Do not remove this line
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+/**
+ * Base component for all widget components in the application.
+ * Provides common functionality for widget components including field translation
+ * and sortable widget containers.
+ *
+ * @class WidgetsComponent
+ * @extends Component
+ * @author Rana Nouman <ranamnouman@gmail.com>
+ */
+export default class WidgetsComponent extends Component {
+	/**
+	 * The internationalization service for handling translations
+	 * @property intl
+	 * @type {Prometheus.Services.intl}
+	 * @private
+	 */
+	@service intl;
+
+	/**
+	 * Processes and formats the fields configuration from widgetSettings
+	 * Translates field labels using the provided translationKey
+	 *
+	 * @property fields
+	 * @type {Array}
+	 * @public
+	 * @returns {Array} Array of field objects with translated labels and value keys
+	 */
+	get fields() {
+		let widgetSettings = this.args.widgetSettings;
+		let translationKey = widgetSettings.translationKey;
+		let fields = [];
+		widgetSettings.fields.forEach((field) => {
+			if (field.label) {
+				fields.push({
+					label: this.intl.t(`${translationKey}.${field.label}`),
+					valueKey: field.valueKey
+				});
+			} else {
+				fields.push({
+					label: this.intl.t(`${translationKey}.${field}`),
+					valueKey: field
+				});
+			}
+		});
+		return fields;
+	}
+
+	/**
+	 * Initializes jQuery UI sortable functionality for widget containers
+	 * Makes widget containers draggable and sortable
+	 *
+	 * @method initSortable
+	 * @public
+	 * @action
+	 */
+	@action
+	initSortable() {
```

Comment:
Weren't we replacing jQuery based component, so if searching is happening using ember, shouldn't sorting also happen using ember?